flocker/restapi/testtools.py

```diff
@@ -179,13 +180,21 @@ class RealTests(mixinClass, TestCase):
         """
         def setUp(self):
             self.app = fixture(self)
-            self.port = reactor.listenTCP(0, Site(self.app.resource()),
-                                          interface="127.0.0.1")
+            path = FilePath(self.mktemp())
+            path.makedirs()
+            self.port = reactor.listenTCP(
+                0, Site(self.app.resource()),
+                interface=b"127.0.0.1",
+            )
+            self.addCleanup(self.port.factory.stopFactory)
```

Comment:
Ehm.  `stopFactory` is called by the reactor (well, via `doStop`) when the factory is no longer listening on a port.  This shouldn't be necessary.